main.scss

```diff
@@ -1,6 +1,10 @@
 // Core styles used for the example pages
+@import "node_modules/govuk-frontend/govuk/base";
```

Comment:
> Another thought on this, do we want to go further and, in future, leave the functional bit of web components uncouples from the styles? We can do so by relying on html attributes instead of classes. Only the JavaScript code would live in this repo and the styles would be moved in the component library along with the generated markup.

I've not really been able to work out myself what I think is the right way to manage this. Doing as you suggest does indeed help resolve the problem of govuk-frontend diverging, however it does seem to create new problems where we'd have two sets of mark-up to keep in sync and releases to co-ordinate.

I think the closest I came to reconcile the correct way to do this would be for this to not rely on any external selectors, so if any other components were embedded they'd have their own class names that are distinct from the govuk-* ones so they don't conflict. Currently this isn't the case though as we depend on govuk-breadcrumbs and I think another component. Though adopting this sort of an approach doesn't feel very compatible with govuk-frontend as a framework as you have to be very delicate to avoid it spilling out any classes of it's own accord. So that just might not be feasible.

Reply:
> it does seem to create new problems where we'd have two sets of mark-up to keep in sync and releases to co-ordinate
a fair concern – I guess the mark-up can be seen as an API for the component itself (for example changing an attribute name would be marked as a breaking change).

> Currently this isn't the case though as we depend on govuk-breadcrumbs and I think another component. Though adopting this sort of an approach doesn't feel very compatible with govuk-frontend as a framework as you have to be very delicate to avoid it spilling out any classes of it's own accord. So that just might not be feasible.

I share the concern and, as you mention in a previous meeting the approach we take on custom elements and `govuk-frontend` is something that we'd like to explore more with the Design System team and figure out a way to move forward while minimising the fear of incompatibility, style leakage or even worse, errors in production.

As mentioned in the review comment, I'm happy to move forward with this approach for de-coupling until we figure out a better one that might involve changes to `govuk-frontend`.